cli/internal/core/engine.go

```diff
@@ -33,13 +33,17 @@ type Engine struct {
 	Tasks            map[string]*Task
 	PackageTaskDeps  map[string][]string
 	rootEnabledTasks util.Set
+
+	// completeGraph is the CompleteGraph. We need this to look up the Pipeline, etc.
+	completeGraph *graph.CompleteGraph
 }
 
 // NewEngine creates a new engine given a topologic graph of workspace package names
-func NewEngine(topologicalGraph *dag.AcyclicGraph) *Engine {
+func NewEngine(completeGraph *graph.CompleteGraph) *Engine {
 	return &Engine{
+		completeGraph:    completeGraph,
 		Tasks:            make(map[string]*Task),
-		TopologicGraph:   topologicalGraph,
+		TopologicGraph:   &completeGraph.WorkspaceGraph,
```

Comment:
This field can be removed and we could access from `completeGraph` directly, but I don't want to go that far in this PR. (See also #3411 which renames this).

Reply:
just so I know / and to learn Go, what could cause a data race?